File: backend/src/main/java/com/marmouset/workout/app/port/in/CreateTrainedExercise.java
package com.marmouset.workout.app.port.in;

import com.marmouset.workout.app.domain.exercise.ExerciseNotFound;
import com.marmouset.workout.app.port.in.dto.CreateTrainedExerciseCommand;
import com.marmouset.workout.app.port.out.dto.TrainedExerciseResponse;

public interface CreateTrainedExercise {
  TrainedExerciseResponse create(CreateTrainedExerciseCommand command)
      throws ExerciseNotFound;
}

File: frontend/src/app/port/in/exercise/CreateTrainedExercise.ts
import TrainedExercise from '../../../domain/exercise/TrainedExercise';
import TrainedExerciseFormDTO from '../dto/TrainedExerciseFormDTO';


export default interface CreateTrainedExercise {
  create(
    workoutId: string, 
    newTrained: TrainedExerciseFormDTO): Promise<TrainedExercise>
}